src/integer_traits.rs

```diff
@@ -0,0 +1,509 @@
+//! Implementation of standard traits like Add, Sub, etc. for the Integer struct
+
+use crate::{integer::IntegerType, value::private::ReprValue, Integer};
+use std::ops::{Add, Div, Mul, Sub};
+
+impl PartialEq for Integer {
+    fn eq(&self, other: &Self) -> bool {
+        match (self.integer_type(), other.integer_type()) {
+            (IntegerType::Fixnum(a), IntegerType::Fixnum(b)) => a.as_rb_value() == b.as_rb_value(),
+            (IntegerType::Bignum(a), IntegerType::Bignum(b)) => unsafe {
+                rb_sys::rb_big_eq(a.as_rb_value(), b.as_rb_value()) == rb_sys::Qtrue.into()
+            },
+            (IntegerType::Fixnum(a), IntegerType::Bignum(b)) => unsafe {
+                let a = rb_sys::rb_int2big(a.to_isize());
+                rb_sys::rb_big_eq(a, b.as_rb_value()) == rb_sys::Qtrue.into()
+            },
+            (IntegerType::Bignum(a), IntegerType::Fixnum(b)) => unsafe {
+                rb_sys::rb_big_eq(a.as_rb_value(), b.as_rb_value()) == rb_sys::Qtrue.into()
+            },
+        }
+    }
+}
+
+impl PartialEq<i64> for Integer {
+    fn eq(&self, other: &i64) -> bool {
+        self.eq(&Integer::from_i64(*other))
+    }
+}
+
+impl PartialEq<Integer> for i64 {
+    fn eq(&self, other: &Integer) -> bool {
+        Integer::from_i64(*self).eq(other)
+    }
+}
```

Comment:
Can we skip all the `impl X<i64> for Interger` and `impl X<Integer> for i64`. Rust doesn't implement these traits across types like this, and we can't match Ruby's full flexibility, so it makes it hard to know how things should behave when we can't match either.